src/care_radiology/api/dicom.py

```
@@ -192,73 +121,45 @@ def upload_with_key(self, request):
             )
 
+    # Link an already-uploaded study to a service request, called once all files for
+    # a study have finished uploading via `upload`/`upload-dicom-external`.
+    @action(detail=False, methods=["post"], url_path="link-service-request")
+    def link_service_request(self, request):
+        service_request_id = request.data.get("service_request_id")
+        study_uid = request.data.get("study_uid")
+
+        service_request = get_object_or_404(ServiceRequest, external_id=service_request_id)
+        if not AuthorizationController.call("can_write_service_request", self.request.user, service_request):
+            raise PermissionDenied(f"You do not have permission to update this service request")
```

Ruff (0.15.20)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)